packages/@expo/cli/src/run/index.ts

```diff
@@ -0,0 +1,80 @@
+#!/usr/bin/env node
+import chalk from 'chalk';
+
+import { Command } from '../../bin/cli';
+import { Log } from '../log';
+import { assertWithOptionsArgs } from '../utils/args';
+import { CommandError, logCmdError } from '../utils/errors';
+import { selectAsync } from '../utils/prompts';
+import { logPlatformRunCommand } from './hints';
+
+export const expoRun: Command = async (argv) => {
+  const args = assertWithOptionsArgs(
+    {
+      // Types
+      '--help': Boolean,
+      '--platform': String,
+      // Aliases
+      '-h': '--help',
+      // All other flags are handled by `expoRunAndroid` or `expoRunIos`
+    },
+    {
+      argv,
+      permissive: true,
+    }
+  );
+
+  if (args['--help']) {
+    Log.exit(
+      chalk`
+  {bold Info}
+    Run the native app locally
+
+  {bold Usage}
+    {dim $} npx expo run:android <dir>
+    {dim $} npx expo run:ios <dir>
+
+  {bold Options}
+    -p, --platform <android|ios>     Run the native app on this platform
+    -h, --help                       Output usage information
+
+    {dim $} npx expo run:android --help    Output Android usage information
+    {dim $} npx expo run:ios --help        Output iOS usage information
+`,
+      0
+    );
+  }
+
+  try {
+    const platform =
+      args['--platform'] ??
+      (await selectAsync('Select the platform to run', [
+        { title: 'Android', value: 'android' },
+        { title: 'iOS', value: 'ios' },
+      ]));
+
+    // Filter `--platform=android|ios` or `--platform android|ios` from the args for `run:android|ios`
+    const argsWithoutPlatform = Object.values(args._).filter(
+      (flag) => !flag.startsWith('--platform') || ['android', 'ios'].includes(flag)
+    );
+
+    logPlatformRunCommand(platform, argsWithoutPlatform);
+
+    switch (platform) {
```

Comment:
The reason we extracted the argument logic in the other commands was so that we could setup unit tests. Given this command is tied to a native build, it isn’t a spectacular candidate for e2e testing.

Reply:
Yeah, seems more like a unit test with some mocking it's probably better suited for.

Reply:
Unfortunately, we can't unit test files using dynamic imports through `await import`. Running into this Jest bug, which seems to be a V8 issue:
https://github.com/aelbore/esbuild-jest/issues/48